src/platform/packages/shared/kbn-unified-metrics-grid/src/components/chart/hooks/use_chart_layers.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import type { LensBaseLayer, LensSeriesLayer } from '@kbn/lens-embeddable-utils/config_builder';
+import type { ChartSectionProps } from '@kbn/unified-histogram/types';
+import useAsync from 'react-use/lib/useAsync';
+import { useMemo } from 'react';
+import type { TimeRange } from '@kbn/data-plugin/common';
+import { getESQLQueryColumns } from '@kbn/esql-utils';
+import { DIMENSIONS_COLUMN } from '../../../common/utils';
+import { useEsqlQueryInfo } from '../../../hooks';
+
+export const useChartLayers = ({
+  query,
+  timeRange,
+  color,
+  seriesType,
+  services,
+  abortController,
+}: {
+  query: string;
+  color?: string;
+  unit?: string;
+  timeRange: TimeRange;
+  seriesType: LensSeriesLayer['seriesType'];
+  services: ChartSectionProps['services'];
+  abortController?: AbortController;
+} & Pick<ChartSectionProps, 'services'>): Array<LensSeriesLayer> => {
+  const queryInfo = useEsqlQueryInfo({ query });
+
+  const { value: columns = [] } = useAsync(
+    () =>
+      getESQLQueryColumns({
+        esqlQuery: query,
+        search: services.data.search.search,
+        signal: abortController?.signal,
+      }),
+
+    [query, services!.expressions, timeRange]
+  );
+
+  const layers = useMemo<LensSeriesLayer[]>(() => {
+    const xAxisColumn = columns.find((col) => col.meta.type === 'date');
+    const xAxis = xAxisColumn?.name ?? '@timestamp';
+
+    const yAxis: LensBaseLayer[] = columns
+      .filter(
+        (col) =>
+          col.name !== DIMENSIONS_COLUMN &&
+          col.meta.type !== 'date' &&
+          !queryInfo.dimensions.some((dim) => dim === col.name)
+      )
+      .map((col) => ({
+        label: col.name,
+        value: col.name,
+        decimals: 1,
+        compactValues: true,
+        seriesColor: color,
```

Comment:
nit: By construction you always have a single metric aggregation in your query, so to make this a bit more "clean" you can:
- add a name for that metric column (as you did for the DIMENSIONS_COLUMN
- use that name and to find/filter the columns and just pick that one.


Is a bit more readable than saying: filter if is not dimension, not date, not the other dimensions.

Also I would suggest to use a bit less generic string for the DIMENSION and the METRIC columns, to avoid any possible collision, something like: `__DIMENSIONS__` or similar